### PR TITLE
Fix possible deadlock on shutdown

### PR DESCRIPTION
**Pull request by myronkscott**

### CODE REVIEW

File: galvan/src/main/java/org/terracotta/testing/master/GalvanStateInterlock.java [hunk 1]
```diff
@@ -339,50 +358,52 @@ public void forceShutdown() throws GalvanFailureException {
...
-        if (null != this.activeServer) {
-          safeStop(this.activeServer);
+        ServerProcess active = null;
+        synchronized (this.sharedLockState) {
+          active = this.activeServer;
+        }
+        if (null != active) {
+          safeStop(active);
         }
         safeWaitWithTimeout(10, TimeUnit.SECONDS);
```
Comment (ramsai1729) [inline galvan/src/main/java/org/terracotta/testing/master/GalvanStateInterlock.java]: This call fails now as no lock is held during the invocation
 ```
 private void safeWaitWithTimeout(long time, TimeUnit units) {
    try {
      this.sharedLockState.wait(units.toMillis(time));
    } catch (InterruptedException e) {
      Assert.unexpected(e);
    }
  }
```

File: galvan/src/main/java/org/terracotta/testing/master/GalvanStateInterlock.java [hunk 2]
```diff
@@ -339,50 +358,52 @@ public void forceShutdown() throws GalvanFailureException {
...
+      while (timeExpired > System.currentTimeMillis() && !checkIfEmpty()) {
+        synchronized (this.sharedLockState) {
+          this.logger.output("* forceShutdown waiting on active: " + (null != this.activeServer)
+              + " passives: " + this.passiveServers.size()
+              + " unknown: " + this.unknownRunningServers.size()
+              + " clients: " + this.runningClients.size()
+              );
+        }
         // Force shut-down all clients, all passives, and the active.
         // (note that we won't modify the collections here, just walk them - we are synchronized)
```
Comment (ramsai1729) [inline galvan/src/main/java/org/terracotta/testing/master/GalvanStateInterlock.java]: remove these comments?

File: galvan/src/main/java/org/terracotta/testing/master/GalvanStateInterlock.java (hunk 1 above)
Comment (myronkscott) [reply galvan/src/main/java/org/terracotta/testing/master/GalvanStateInterlock.java]: fixed

File: galvan/src/main/java/org/terracotta/testing/master/GalvanStateInterlock.java (hunk 2 above)
Comment (myronkscott) [reply galvan/src/main/java/org/terracotta/testing/master/GalvanStateInterlock.java]: fixed

**Pull request merged**